app/lib/report_helper.rb

```diff
@@ -927,6 +936,26 @@ def flat_hash(h, f = [], g = {})
     # into    {[:a, :b, :c] => 1, [:a, :b, :d] => 2, [:a, :e] => 3, [:f] => 4}
   end
 
+  def values_in_hash_keys(input)
+    # Results from some SQL queries
+    res = []
+    input.each do |k, _|
+      k.each do |_, id|
+        res << id
+      end
+    end
+    res
+  end
+
+  def only_species_or_genus_counts!(tax_2d, species_selected)
```

Comment:
It seems a bit excessive here. Isn't the difference between specieis_level_counts and genus_level_acounts just checking tax_level. 
why not have a function do
select_proper_level_counts(tax_2d, TaxonCounts::LEVEL_GENUS|SPECIES) etc